segregation/_base.py

```diff
@@ -217,12 +217,14 @@ def __init__(self, w, network, distance=1000, decay='linear', function='triangul
                 decay=decay,
                 precompute=precompute,
             )
+            self._original_data = self.data.copy()
             self.data = access
             self.network = network
         elif w:
             self.data = _build_local_environment(self.data, self._groups, w, function=function)
             self.w = w
         elif distance and not network:
+            self._original_data = self.data.copy()
```

Comment:
What's the reason for creating an identically named data variable? Same comment above and below.

Reply:
here the copy is definitely necessary. We want to store the input data in the new class but need to make sure we have a copy rather than a pointer to the original

Reply:
(for most of these copies i discovered the need through trial and error after accidentially modifying the inuput...)

Reply:
Yep, I got a bit trigger happy here. 😄